webapp/channels/src/components/admin_console/schema_admin_settings.tsx

```diff
@@ -33,28 +32,60 @@ import Tooltip from 'components/tooltip';
 import WarningIcon from 'components/widgets/icons/fa_warning_icon';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 import AdminHeader from 'components/widgets/admin_console/admin_header';
-
 import Setting from './setting';
 
 import './schema_admin_settings.scss';
+import {Role} from '@mattermost/types/roles';
+import {AdminConfig, ClientConfig, EnvironmentConfig} from '@mattermost/types/config';
+import {CloudState} from '@mattermost/types/cloud';
+import {DeepPartial} from '@mattermost/types/utilities';
+import {InputTypes} from 'components/widgets/settings/text_setting';
+
+type ConsoleAccess = {
+    read: Record<string, boolean>;
+    write: Record<string, boolean>;
+}
 
-export default class SchemaAdminSettings extends React.PureComponent {
-    static propTypes = {
-        config: PropTypes.object,
-        environmentConfig: PropTypes.object,
-        setNavigationBlocked: PropTypes.func,
-        schema: PropTypes.object,
-        roles: PropTypes.object,
-        license: PropTypes.object,
-        editRole: PropTypes.func,
-        updateConfig: PropTypes.func.isRequired,
-        isDisabled: PropTypes.bool,
-        consoleAccess: PropTypes.object,
-        cloud: PropTypes.object,
-        isCurrentUserSystemAdmin: PropTypes.bool,
-    };
+type Props = {
+    schema: any;
+    setNavigationBlocked?: (blocked: boolean) => void;
+    isDisabled?: boolean;
+    consoleAccess?: ConsoleAccess;
+    enterpriseReady?: boolean;
+    license?: Record<string, any>;
+    config?: DeepPartial<AdminConfig>;
+    clientConfig?: ClientConfig;
+    environmentConfig?: Partial<EnvironmentConfig>;
+    roles?: Record<string, Role>;
+    editRole?: (role: Role) => void;
+    updateConfig?: any;
+    cloud?: CloudState;
+    isCurrentUserSystemAdmin?: boolean;
+}
+
+type Settings = {
+    [key: string]: any;
+}
+type State = {
+    saveNeeded: boolean | string;
+    saving: boolean;
+    serverError: null;
+    errorTooltip: boolean;
+    customComponentWrapperClass: string;
+    confirmNeededId: string;
+    showConfirmId: string;
+    clientWarning: string;
+    prevSchemaId?: string;
+    [key: string]: any;
+}
 
-    constructor(props) {
+export default class SchemaAdminSettings extends React.PureComponent<Props, State> {
+    buildSettingFunctions: any;
+    errorMessageRef: any;
+    handleSaved: any;
```

Comment:
we still seems to have some more any lefts, can you please take a look.

Reply:
@M-ZubairAhmed Removed Any's with specific types